light/client.go

```diff
@@ -151,14 +151,29 @@ type Client struct {
 	logger log.Logger
 }
 
+func validatePrimaryAndWitnesses(primary provider.Provider, witnesses []provider.Provider) error {
+	witnessMap := make(map[string]struct{})
+	for _, w := range witnesses {
+		if w.ID() == primary.ID() {
+			return fmt.Errorf("primary cannot be also configured as witness")
```

Comment:
Does this imply that it is now impossible to run a light client against a network with only one node (e.g., as we did to repro the stall issue)?

Reply:
As i mentioned above, it is possible to run it without `--witness` and in this case we skip the header check against the witness.

Reply:
Oh yes, not specifying `--witnesses` is only possible in this PR branch.